modules/optimization/examples/diffusion_reaction/forward_exact.i

```diff
@@ -0,0 +1,152 @@
+[Mesh]
+  [square]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 16
+    ny = 16
+    xmin = 0
+    xmax = 1
+    ymin = 0
+    ymax = 1
+  []
+[]
+
+[Variables/u]
+[]
+
+[Functions]
+  [rxn_func]
+    type = ParsedFunction
+    expression = 'exp(x * y) - 1'
+  []
+[]
+
+[Materials]
+  [ad_dc_prop]
+    type = ADParsedMaterial
+    expression = '1 + u'
+    coupled_variables = 'u'
+    property_name = dc_prop
+  []
+  [ad_rxn_prop]
+    type = ADGenericFunctionMaterial
+    prop_values = 'rxn_func'
+    prop_names = rxn_prop
+    outputs = exodus
+  []
+  [ad_neg_rxn_prop]
+    type = ADParsedMaterial
+    expression = '-rxn_prop'
+    material_property_names = 'rxn_prop'
+    property_name = 'neg_rxn_prop'
+  []
```

Comment:
Why do we need the positive and the negative reaction rate?

Reply:
[MatReaction](https://mooseframework.inl.gov/source/kernels/ADMatReaction.html) includes a negative sign in residual evaluation, so we need to reverse this with a negative reaction rate. However, I wanted the parameter to remain positive, which is why there is one object to evaluate function and another to flip it's sign for the kernel.

Reply:
I added this Zachs reply to the input files.